test/unit/oidc_jwt_token_validation_test.rb

```diff
@@ -0,0 +1,49 @@
+require 'test_helper'
+require 'ostruct'
+
+class OidcJwtTokenValidationTest < ActiveSupport::TestCase
+
+  # Jwt token created with the folloing payload:
+  # { "name": "jwt token", "iat": 1557224758, "exp": 1557228982, "typ": "Bearer", "aud": "rest-client"}
+  let(:token) { 'eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoxMjMsImlhdCI6MTUxNDgwNDQwMCwianRpIjoiM2U4Mjg2OTQwZWIxNjJlYzczNWY4YTVhYWM1OTI2MDM3ZmRjN2YwNWU1MjFhNzQyMzFhNjVmMjU1N2E4YWY5NCJ9.TSM2xMnMuMEWwAVoIidyLIwJElJQZVQvcfaxyVA7cDI' }
+
+  test 'if all checks pass for valid_jwt?' do
```

Comment:
I had tried to stub the JWT.decode method using minitest but I failed to do so because of the decode method calls [generate_jwks_url](https://github.com/theforeman/foreman/pull/6549/files#diff-7c015c7ad0f51b2d74434a4730c5e751R40) that needs an URL and thus I always get a validation error.

I tried to do something like:
```
JWT.stubs(:decode).returns([{ "name": "jwt token", "iat": 1557224758, "exp": 1557228982, "typ": "Bearer", "aud": "rest-client"}])

and

JWT.expects(:decode).returns([{ "name": "jwt token", "iat": 1557224758, "exp": 1557228982, "typ": "Bearer", "aud": "rest-client"}])
```

Both gave the same validation error asking for the URL which is present in the [generate_jwks_uri](https://github.com/theforeman/foreman/pull/6549/files#diff-7c015c7ad0f51b2d74434a4730c5e751R29)

@timogoebel @tbrisker @ares  Can you please suggest me a way to test these methods in a better way?

In RSpec i could easily do something like: 
```
allow(JWT).to receive(:decode).and_return(:value)
```

Reply:
The stubbing code for minitest looks correct. This might be another issue?!